fuel/app/classes/controller/widgets.php

```diff
@@ -239,101 +240,92 @@ public function get_preview_widget($inst_id, $is_embedded = false)
 	protected function show_editor($title, $widget, $inst_id=null)
 	{
 		$this->_disable_browser_cache = true;
-		Css::push_group(['core', 'widget_create']);
-		Js::push_group(['angular', 'materia', 'author']);
-		if ( ! empty($widget->creator) && preg_match('/\.swf$/', $widget->creator))
-		{
-			// add swfobject if it's needed
-			Js::push_group('swfobject');
-		}
 
+		Js::push_inline('var WIDGET_HEIGHT = "'.$widget->height.'";');
+		Js::push_inline('var WIDGET_WIDTH = "'.$widget->width.'";');
+
+
+		$this->theme = Theme::instance();
+		$this->theme->set_template('layouts/react');
 		$this->theme->get_template()
 			->set('title', $title)
-			->set('page_type', 'create');
+			->set('page_type', 'widget');
 
-		$this->theme->set_partial('footer', 'partials/angular_alert');
-		$this->theme->set_partial('content', 'partials/widget/create')
-			->set('widget', $widget)
-			->set('inst_id', $inst_id);
+		Css::push_group(['createpage', 'core', 'widget_create']);
+		Js::push_group(['react', 'createpage']);
 	}
 
 	protected function draft_not_playable()
 	{
 		$this->_disable_browser_cache = true;
+
+		$this->theme = Theme::instance();
+		$this->theme->set_template('layouts/react');
 		$this->theme->get_template()
 			->set('title', 'Draft Not Playable')
 			->set('page_type', '');
 
-		$this->theme->set_partial('content', 'partials/widget/draft_not_playable');
-		$this->theme->set_partial('footer', 'partials/angular_alert');
-
-		Js::push_group(['angular', 'materia']);
+		Js::push_group(['react', 'draft_not_playable']);
 	}
 
 	protected function retired()
 	{
+		$this->theme = Theme::instance();
+		$this->theme->set_template('layouts/react');
 		$this->theme->get_template()
 			->set('title', 'Retired Widget')
 			->set('page_type', '');
 
-		$this->theme->set_partial('content', 'partials/widget/retired');
-		$this->theme->set_partial('footer', 'partials/angular_alert');
-
-		Js::push_group(['angular', 'materia']);
+		Js::push_group(['react', 'retired']);
 	}
 
 	protected function no_attempts(object $inst, bool $is_embedded)
 	{
 		$this->_disable_browser_cache = true;
+		$this->theme = Theme::instance();
+		$this->theme->set_template('layouts/react');
 		$this->theme->get_template()
 			->set('title', 'Widget Unavailable')
 			->set('page_type', 'login');
 
-		$this->theme->set_partial('footer', 'partials/angular_alert');
-		$this->theme->set_partial('content', 'partials/widget/no_attempts')
-			->set('classes', 'widget')
-			->set('attempts', $inst->attempts)
-			->set('scores_path', '/scores'.($is_embedded ? '/embed' : '').'/'.$inst->id)
-
-			->set('summary', $this->theme->view('partials/widget/summary')
-				->set('type',$inst->widget->name)
-				->set('name', $inst->name)
-				->set('icon', Config::get('materia.urls.engines')."{$inst->widget->dir}img/icon-92.png"));
+		Js::push_inline('var WIDGET_HEIGHT = "'.$widget->height.'";');
```

Comment:
`$widget` is undefined here. You probably meant `$inst->widget->height`?

Reply:
Oh, yep. The WIDGET_HEIGHT variable isn't being used, though, so I'll just remove it.